pkg/workflow/engine/model/workflow/workflow.go

```diff
@@ -0,0 +1,53 @@
+// Copyright 2020 Chaos Mesh Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package workflow
+
+import (
+	"github.com/chaos-mesh/chaos-mesh/pkg/workflow/engine/model/node"
+	"github.com/chaos-mesh/chaos-mesh/pkg/workflow/engine/model/template"
+)
+
+type WorkflowSpec interface {
```

Comment:
How we validate the workflow spec?

Reply:
Any implementation of `WorkflowSpec` should guarantee correctness by itself. 

As an implementation with Kubernetes, it should be resolved by webhook. (WIP)